app/controllers/concerns/cookie_preference_concerns.rb

```diff
@@ -5,7 +5,7 @@ def create
     session[:display_cookie_consent_confirmation] = true
 
     flash[:success] = 'Your cookie preferences have been updated.'
-    redirect_back(fallback_location: cookie_page_path)
+    redirect_to session[:previous_referer].presence || root_path
```

Comment:
This was just re-rending the cookie page as that was the request.referer. We set the session[:previous_referrer] in the `cookies_page` method in both the relevant provider and candidate controllers, but then didn't use them for the redirect.